src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -895,15 +955,29 @@ abstract class ManagedStrategy(
         }
 
         private fun onNextThreadSwitchPoint(nextThread: Int) {
+            /*
+                When we're back to some thread, newSwitchPoint won't be called before the fist
+                in current thread part as it was called before switch.
```

Comment:
Please re-read your changes and comments slowly, so you can polish them. Then, the PR should be ready to go.